rewrite-java/src/main/java/org/openrewrite/java/marker/JavaVersion.java

```diff
@@ -43,4 +34,57 @@
     String vmVendor;
     String sourceCompatibility;
     String targetCompatibility;
+
+    public static MajorVersion getMajorVersion(String version) {
+        return MajorVersion.get(version);
+    }
+
+    public enum MajorVersion {
+        INVALID("invalid"),
+        JAVA_6("6"),
```

Comment:
Why this enum?

Reply:
I waited for feedback on the enum before adding another constructor to UsesJavaVersion.

We can have a method like `String getMajorVersion(String)` that normalizes the input, but I think an enum is clearer for recipe authors: `new UsesJavaVersion(JavaVersion.MajorVersion.Java_8)`.

Do you have concerns about an enum?

Reply:
Why not just an integer? `8`, `11`? Just feels like bloat with little benefit.

Reply:
hah, now looking at the enum in the context of this discussion, An integer makes quite a lot of sense.

Reply:
Updated to use an int